bundles/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/handler/IPBridgeThingHandler.java

```diff
@@ -70,20 +110,64 @@ public void initialize() {
         String ip = config.getIpAddress();
         InetSocketAddress localEndPoint = null;
         boolean useNAT = false;
-        int ipConnectionType;
+
+        IPClient.IpConnectionType ipConnectionType;
         if (MODE_TUNNEL.equalsIgnoreCase(connectionTypeString)) {
             useNAT = config.getUseNAT();
-            ipConnectionType = CustomKNXNetworkLinkIP.TUNNELING;
+            ipConnectionType = IPClient.IpConnectionType.TUNNEL;
+        } else if (MODE_SECURE_TUNNEL.equalsIgnoreCase(connectionTypeString)) {
+            useNAT = config.getUseNAT();
+            ipConnectionType = IPClient.IpConnectionType.SECURE_TUNNEL;
+
+            if (!securityAvailable) {
+                logger.warn("Bridge {} missing security configuration for secure tunnel", thing.getUID());
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "Security configuration missing for secure tunnel");
+                return;
+            }
+            boolean tunnelOk = ((secureTunnel.user > 0) && (secureTunnel.devKey.length == 16)
+                    && (secureTunnel.userKey.length == 16));
+            if (!tunnelOk) {
+                logger.warn("Bridge {} incomplete security configuration for secure tunnel", thing.getUID());
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "Security configuration for secure tunnel is incomplete");
+                return;
+            }
+
+            logger.info("KNX secure tunneling needs a few seconds to establish connection");
+            // user id, key, devAuth are already stored
         } else if (MODE_ROUTER.equalsIgnoreCase(connectionTypeString)) {
             useNAT = false;
             if (ip.isEmpty()) {
                 ip = KNXBindingConstants.DEFAULT_MULTICAST_IP;
             }
-            ipConnectionType = CustomKNXNetworkLinkIP.ROUTING;
+            ipConnectionType = IPClient.IpConnectionType.ROUTER;
+        } else if (MODE_SECURE_ROUTER.equalsIgnoreCase(connectionTypeString)) {
+            useNAT = false;
+            if (ip.isEmpty()) {
+                ip = KNXBindingConstants.DEFAULT_MULTICAST_IP;
+            }
+            ipConnectionType = IPClient.IpConnectionType.SECURE_ROUTER;
+
+            if (!securityAvailable) {
+                logger.warn("Bridge {} missing security configuration for secure routing", thing.getUID());
```

Comment:
Could you add specific checks for the parameters, so that you could directly hint to the user, which parameter he forgot to provide?

Reply:
check for is directly below for secure router.
this if is just for the case when security is not configured at all....